thirdeye/thirdeye-frontend/app/pods/rootcause/route.js

```diff
@@ -45,31 +59,49 @@ const _filterToUrn = (filters) => {
 
 export default Ember.Route.extend(AuthenticatedRouteMixin, {
   queryParams: {
-    granularity: queryParamsConfig,
-    filters: queryParamsConfig,
-    compareMode: queryParamsConfig,
-    analysisRangeStart: queryParamsConfig,
-    analysisRangeEnd: queryParamsConfig,
-    anomalyRangeStart: queryParamsConfig,
-    anomalyRangeEnd: queryParamsConfig
+    metricId: queryParamsConfig,
+    anomalyId: queryParamsConfig,
+    shareId: queryParamsConfig
   },
 
   model(params) {
-    const { rootcauseId: id } = params;
+    const { metricId, anomalyId, shareId } = params;
 
-    // TODO handle error better
-    if (!id) { return; }
+    let metricUrn;
+    if (metricId) {
+      metricUrn = `thirdeye:metric:${metricId}`;
+    }
+
+    let anomalyUrn;
+    if (anomalyId) {
+      anomalyUrn = `thirdeye:event:anomaly:${anomalyId}`;
+    }
+
+    let share;
+    if (shareId) {
+      share = fetch(`/config/rootcause-share/${shareId}`).then(res => res.json());
+    }
+
+    let anomalyContext;
```

Comment:
let's try to define variables at the beginning of this function scope (model)

Reply:
fixed